build/dom_utils.js

```diff
@@ -206,16 +209,58 @@ function fireEvent(eventName, target) {
   target.dispatchEvent(event);
 };
 
+var patchRefs = function patchRefs(refs) {
+  var promises = Object.keys(refs).reduce(function (acc, key) {
+    var $el = refs[key];
+    if ($el instanceof Array) {
+      return acc.concat($el);
+    }
+    return acc.concat($el);
+  }, []).map(function ($el) {
+    return $el.whenComponentConnected = new Promise(function (resolve, reject) {
+      $el.__whenConnectedResolver = function (arg) {
+        return resolve(arg || $el);
+      };
+    });
+  });
+
+  return refs;
+};
+
 /**
- *  Set the HTML content of element, or generate DocumentFragment
- *  @param {HTMLElement|String} target Element to set content or html string
- *  @param {String|HTMLElement|DocumentFragment} [content] HTML content string
- *  @return {Array|Function} tuple [target || DocumentFragment, refs object] or render function
+ * Set the HTML content of element, or generate DocumentFragment
+ * @param {HTMLElement|String} target Element to set content or html string
+ * @param {String|HTMLElement|DocumentFragment} [content] HTML content string
+ * @return {Array|Function} tuple [target || DocumentFragment, refs object] or render function
+ *
+ * @example
+ * // set content of element
+ * const $el = document.createElement('div');
+ * const result = html($el, `<div></div>`); //=> [$el, {}];
+ *
+ * @example
+ * // create element renderer
+ * const $el = document.createElement('div');
+ * const render = html($el);
+ * const result = render(`<div></div>`); //=> [$el, {}];
+ *
+ * @example
+ * // generate document fragment from string
+ * const result = html(`<div id="some_id"></div>`) //=> [<HTMLElement#some_id>, {}]
+ *
+ * @example
+ * // get refs
+ * const refs = html(`
+ *   <div id="some_id" ref="some_ref">
+ *     <span refs="span_elements" id="span_1"></span>
+ *     <span refs="span_elements" id="span_2"></span>
+ *   </div>
+ * `)[1] //=> { some_ref: <HTMLElement#some_id>, span_elements: [<HTMLElement#span_1>, <HTMLElement#span_2>] }
  * */
 function html(target, content) {
   var fn = function fn(content) {
```

Comment:
As everything is passed through `babel`; we can use `const`/`let` and also the arrow functions.

Reply:
And Node.js LTS version is 9.x; we can expect that support to be present

Reply:
it's build. actual code is in src